### DISCUSSION TITLE
can nicegui be packaged into a single executable

### DISCUSSION THREAD

**Discussion by laoshaw** [zauberzeug/nicegui#394] [Q&A] [answered]
will pyinstaller work well with nicegui, or it's really for running in a python envrionment.

how about Qt libraries, to deploy a single nicegui app the Qt libraries has to be loaded dynamically, how to do that?

what's the advantage over tkinter, yes tkinter is not very appealing but it still can do GUI well

**Comment by rodja** [comment, marked as answer]
Yes, NiceGUI works with PyInstaller (see #355). Thoughts about Qt (or similar) to run a desktop app are bundled in #89. The main advantage over tkinter is that NiceGUI runs in your browser (besides nicer APIs and looks of course). The web-based solution allows multiple users, remote access and usage of all the web libraries out there.
